src/pieces/queen.js

```diff
@@ -6,8 +6,38 @@ class Queen extends Piece {
     this.name = 'queen';
     this.display = `<i class="fas fa-chess-queen ${side}"></i>`; //fontawesome queen
   }
-  findLegalMoves() {
+  findLegalMoves(board) {
     const possibleMoves = [];
+    const directions = [
+      [0, -1], // left
+      [0, 1], // right
+      [-1, 0], // down
+      [1, 0], // up
+      [1, 1], // up right
+      [-1, -1], // down left
+      [1, -1], // up left
+      [-1, 1], // down right
+    ];
+
+    directions.forEach((directions) => {
+      for (let distance = 1; distance < 8; distance++) {
+        const newRow = this.row + distance * directions[0];
+        const newColumn = this.column + distance * directions[1];
+
+        if (newRow >= 0 && newRow <= 7 && newColumn >= 0 && newColumn <= 7) {
+          const targetSquare = board.getSquare(newRow, newColumn);
+
+          if (!targetSquare.piece) {
+            possibleMoves.push([newRow, newColumn]);
+          } else if (targetSquare.piece.side !== this.side) {
+            possibleMoves.push([newRow, newColumn]);
+            break;
+          } else {
+            break;
+          }
```

Comment:
To jest git, ja osobiście zapisałbym to troszkę inaczej:
```suggestion
          if (!targetSquare.piece) {
            possibleMoves.push([newRow, newColumn]);
            continue;
          }

          if (targetSquare.piece?.side !== this.side) {
            possibleMoves.push([newRow, newColumn]);
          }

          break;
```

Reply:
To jest bardziej opinia niż reguła (ale dość popularna) cieżko się czyta kod który korzysta  z `else` bo musisz cały czas pamiętać co było w poprzednim `ifie` , dlatego ja osobiście zalecam unikanie słowa kluczowego else w kodzie